### PR TITLE
Added QR code component

### PR DESCRIPTION
**Pull request by sauravmaiti22**
Added QR code component

### CODE REVIEW

**Comment by donalnofrixion** [comment]
The QR code should be able to be downloaded as per the Figma.

File: src/components/ui/QRCode/QRCode.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import QRCodeComponent from 'react-qr-code';
```
Comment (donalnofrixion) [inline src/components/ui/QRCode/QRCode.tsx:1]: The QR code should be able to be downloaded as per the Figma.

**Review by donalnofrixion — CHANGES_REQUESTED** [review]

**Comment by sauravmaiti22** [comment]
> The QR code should be able to be downloaded as per the Figma.

Didn't see that in edit mode. Will add it now.

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Hey @sauravmaiti22, I could download the QR, but it seems to be the same image we can see on the HTML. The idea for this would be to download a bigger image that can be used in other contexts, like printing and so on. 

I think at least we would need an image at 1024x1024px.

File: src/components/ui/QRCode/QRCode.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { AnimatePresence, motion } from 'framer-motion';
+
+const QRCode = ({ url }: { url: string }) => {
+  const [isHovered, setIsHovered] = useState(false);
+  const qrCodeRef = useRef<HTMLDivElement>(null);
+
+  const downloadAsImage = () => {
+    const element = qrCodeRef.current as HTMLDivElement;
+
+    html2canvas(element).then((canvas) => {
```
Comment (donalnofrixion) [inline src/components/ui/QRCode/QRCode.tsx:13]: We should think about giving the downloaded QR file a meaningful name. Something that tells you what it is. How much, etc

**Review by donalnofrixion — CHANGES_REQUESTED** [review]

File: src/components/ui/QRCode/QRCode.tsx (hunk 2 above)
Comment (sauravmaiti22) [reply src/components/ui/QRCode/QRCode.tsx:13]: Yes. I thought about it but it was getting really long. One suggestion would be nf-PR-{PaymentReqestID}. Adding amount in the name would be tricky with decimal.

Comment (donalnofrixion) [reply src/components/ui/QRCode/QRCode.tsx:13]: It would be worth ask Andres and Pablo what they think.

Comment (sauravmaiti22) [reply src/components/ui/QRCode/QRCode.tsx]: Updated PR. Changed the file name format to QR-DD-MM-YYYY-HHMMSS.

**Comment by sauravmaiti22** [comment]
> Hey @sauravmaiti22, I could download the QR, but it seems to be the same image we can see on the HTML. The idea for this would be to download a bigger image that can be used in other contexts, like printing and so on.
> 
> I think at least we would need an image at 1024x1024px.



> Hey @sauravmaiti22, I could download the QR, but it seems to be the same image we can see on the HTML. The idea for this would be to download a bigger image that can be used in other contexts, like printing and so on.
> 
> I think at least we would need an image at 1024x1024px.

@pabloi-nf I have made the change to download the QR code in a bigger size.

**Review by donalnofrixion — APPROVED** [review]

**Review by Publicker — APPROVED** [review]

**Review by pabloi-nf — APPROVED** [review]

**Pull request merged**